contracts/modules/cover/Cover.sol

```diff
@@ -105,11 +108,13 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon, ReentrancyGuard, Mu
     PoolAllocationRequest[] memory poolAllocationRequests
   ) external payable onlyMember nonReentrant whenNotPaused returns (uint coverId) {
 
-    if (params.period < MIN_COVER_PERIOD) {
+    // check only for a new cover
```

Comment:
the duration of the resulting segment should also be between min / max period. 

i.e. if your cover has 5 days left, you can extend it by at least 23 days, so the total period of the new segment is at least 28 days. same for max